apis/datadoghq/common/v1/agent_types.go

```diff
@@ -36,3 +39,65 @@ type AgentImageConfig struct {
 	// +optional
 	PullSecrets *[]corev1.LocalObjectReference `json:"pullSecrets,omitempty"`
 }
+
+// DaemonSetStatus defines the observed state of Agent running as DaemonSet.
+// +k8s:openapi-gen=true
+// +kubebuilder:object:generate=true
+type DaemonSetStatus struct {
+	Desired   int32 `json:"desired"`
+	Current   int32 `json:"current"`
+	Ready     int32 `json:"ready"`
+	Available int32 `json:"available"`
+	UpToDate  int32 `json:"upToDate"`
+
+	Status      string       `json:"status,omitempty"`
+	State       string       `json:"state,omitempty"`
+	LastUpdate  *metav1.Time `json:"lastUpdate,omitempty"`
+	CurrentHash string       `json:"currentHash,omitempty"`
+
+	// DaemonsetName corresponds to the name of the created DaemonSet.
+	DaemonsetName string `json:"daemonsetName,omitempty"`
+}
+
+// DeploymentStatus type representing a Deployment status.
+// +k8s:openapi-gen=true
+// +kubebuilder:object:generate=true
+type DeploymentStatus struct {
+	// Total number of non-terminated pods targeted by this deployment (their labels match the selector).
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
+	// +optional
+	UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`
+
+	// Total number of ready pods targeted by this deployment.
+	// +optional
+	ReadyReplicas int32 `json:"readyReplicas,omitempty"`
+
+	// Total number of available pods (ready for at least minReadySeconds) targeted by this deployment.
+	// +optional
+	AvailableReplicas int32 `json:"availableReplicas,omitempty"`
+
+	// Total number of unavailable pods targeted by this deployment. This is the total number of
+	// pods that are still required for the deployment to have 100% available capacity. They may
+	// either be pods that are running but not yet available or pods that still have not been created.
+	// +optional
+	UnavailableReplicas int32 `json:"unavailableReplicas,omitempty"`
+
+	LastUpdate  *metav1.Time `json:"lastUpdate,omitempty"`
+	CurrentHash string       `json:"currentHash,omitempty"`
+
+	// GeneratedToken corresponds to the generated token if any token was provided in the Credential configuration when ClusterAgent is
+	// enabled.
+	// +optional
+	GeneratedToken string `json:"generatedToken,omitempty"`
+
+	// Status corresponds to the ClusterAgent deployment computed status.
```

Comment:
DeploymentStatus can be use also for the clustercheckrunner so it can be good to remove the notion of ClusterAgent in the description